allauth/account/adapter.py

```diff
@@ -503,7 +504,14 @@ def is_safe_url(self, url):
                 is_safe_url as url_has_allowed_host_and_scheme,
             )
 
-        return url_has_allowed_host_and_scheme(url, allowed_hosts=None)
+        # get_host already validates the given host, so no need to check it again
+        allowed_hosts = {self.request.get_host()} | set(settings.ALLOWED_HOSTS)
+
+        if "*" in allowed_hosts:
+            parsed_host = urlparse(url).netloc
```

Comment:
If `url` is relative,`parsed_host` will be the empty string. In that case, shouldn't `allowed_hosts` be `None`?

Reply:
Looking at the source code for `url_has_allowed_host_and_scheme`, there's a check for `not url_info.netloc` which will return true, see here: https://github.com/django/django/blob/main/django/utils/http.py#L298

In this case it doesn't matter if allowed_hosts is None or a set because django assumes relative URLs are safe.

Reply:
Agreed, though if we write it like `allowed_hosts={parsed_host} if parsed_host else None` at least the code is understandable without having to rely on the inner workings of the Django code.

Reply:
fair enough, will make that change now!